llvm/test/Transforms/LoopVectorize/AArch64/partial-reduce-dot-product.ll

```diff
@@ -0,0 +1,1455 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 4
+; RUN: opt -passes=loop-vectorize -force-vector-interleave=1 -enable-epilogue-vectorization=false -force-target-instruction-cost=1 -S < %s | FileCheck %s --check-prefixes=CHECK-INTERLEAVE1
+; RUN: opt -passes=loop-vectorize -force-target-instruction-cost=1 -enable-epilogue-vectorization=false -S < %s | FileCheck %s --check-prefixes=CHECK-INTERLEAVED
```

Comment:
Do the tests need force-target-instruction-cost=1 or will the cost model make the right choice? If so, would probably be better to exercise the cost model for more cases

Reply:
They do need this until the max vector bandwidth patch is re-landed. I will remove this once that has happened.

Reply:
Ah I see, although wouldn't it be sufficient in that case to pass `-vectorizer-maximize-bandwidth` assuming the patch I am thinking of is https://github.com/llvm/llvm-project/pull/109671/files?

Reply:
If I remove the instruction cost option then we only see scalable vectors so I think we should keep it in place to test fixed-width vectors, and once the bandwidth PR is re-landed I'll make some adjustments to the cost model to test scalable and fixed.

Reply:
Ah right, there are 2 cases to cover IIUC, 1) NEON only (SVE not available), this should cover NEON cost-modeling, and 2) SVE available which test picking between NEON or SVE.

Could we have a separate test that doesn't have +sve, possibly in a separate file to test fixed-with codegen, without needing `-force-target-instruction-cost=1`?

Reply:
There's now a NEON test, do we still need  `-force-target-instruction-cost=1` here?